LinkByName with an altname that exceeds the 15 char limit fails

The test for altnames only contains short (<15 char) altnames.
altnames are allowed to carry more characters. See:

```bash
# Query via Name
mava@server01:~/projects$ ip l show dev foobar
22: foobar: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN mode DEFAULT group default qlen 1000
    link/ether 72:42:33:ae:b7:bf brd ff:ff:ff:ff:ff:ff
    altname some_longer_altname

# Query via long AltName
mava@server01:~/projects$ ip l show dev vx-some_very_long_node_name_l1_e1-1
22: foobar: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN mode DEFAULT group default qlen 1000
    link/ether 72:42:33:ae:b7:bf brd ff:ff:ff:ff:ff:ff
    altname some_longer_altname
```
However, using this library I am getting the following error:

```bash
mava@server01:~/projects/netlink$ sudo go test -timeout 30s -run ^TestLinkAltName$ github.com/vishvananda/netlink
--- FAIL: TestLinkAltName (0.00s)
    link_test.go:1937: numerical result out of range: Attribute failed policy validation
FAIL
FAIL    github.com/vishvananda/netlink  0.005s
FAIL
```
Find the adjusted test here:
https://github.com/steiler/netlink/commit/32fec641ca0b95078a2e620ba6b17023c0efac02

@SchSeba Maybe you can take a look at this. After implementing #862

--- a/link_linux.go
+++ b/link_linux.go
@@ -1867,6 +1867,9 @@ func (h *Handle) LinkByName(name string) (Link, error) {
 	req.AddData(attr)
 
 	nameData := nl.NewRtAttr(unix.IFLA_IFNAME, nl.ZeroTerminated(name))
+	if len(name) > 15 {
+		nameData = nl.NewRtAttr(unix.IFLA_ALT_IFNAME, nl.ZeroTerminated(name))
+	}
 	req.AddData(nameData)
 
 	link, err := execGetLink(req)

--- a/link_test.go
+++ b/link_test.go
@@ -10,6 +10,8 @@ import (
 	"net"
 	"os"
 	"os/exec"
+	"sort"
+	"strings"
 	"syscall"
 	"testing"
 	"time"
@@ -1899,14 +1901,15 @@ func TestLinkAltName(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	err = LinkAddAltName(link, "altname")
-	if err != nil {
-		t.Fatalf("Could not add altname: %v", err)
-	}
+	altNames := []string{"altname", "altname2", "some_longer_altname"}
+	sort.Strings(altNames)
+	altNamesStr := strings.Join(altNames, ",")
 
-	err = LinkAddAltName(link, "altname2")
-	if err != nil {
-		t.Fatalf("Could not add altname: %v", err)
+	for _, altname := range altNames {
+		err = LinkAddAltName(link, altname)
+		if err != nil {
+			t.Fatalf("Could not add %s: %v", altname, err)
+		}
 	}
 
 	link, err = LinkByName("bar")
@@ -1914,49 +1917,40 @@ func TestLinkAltName(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	altNameExist := false
-	altName2Exist := false
-	for _, altName := range link.Attrs().AltNames {
-		if altName == "altname" {
-			altNameExist = true
-		} else if altName == "altname2" {
-			altName2Exist = true
-		}
+	sort.Strings(link.Attrs().AltNames)
+	linkAltNamesStr := strings.Join(link.Attrs().AltNames, ",")
 
-	}
-	if !altNameExist {
-		t.Fatal("Could not find altname")
+	if altNamesStr != linkAltNamesStr {
+		t.Fatalf("Expected %s AltNames, got %s", altNamesStr, linkAltNamesStr)
 	}
 
-	if !altName2Exist {
-		t.Fatal("Could not find altname2")
-	}
-
-	link, err = LinkByName("altname")
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	err = LinkDelAltName(link, "altname")
-	if err != nil {
-		t.Fatalf("Could not delete altname: %v", err)
-	}
-
-	link, err = LinkByName("bar")
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	altNameExist = false
-	for _, altName := range link.Attrs().AltNames {
-		if altName == "altname" {
-			altNameExist = true
-			break
+	for _, altname := range altNames {
+		link, err = LinkByName(altname)
+		if err != nil {
+			t.Fatal(err)
 		}
 	}
-	if altNameExist {
-		t.Fatal("altname still exist")
+
+	for idx, altName := range altNames {
+		err = LinkDelAltName(link, altName)
+		if err != nil {
+			t.Fatalf("Could not delete %s: %v", altName, err)
+		}
+
+		link, err = LinkByName("bar")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		sort.Strings(link.Attrs().AltNames)
+		linkAltNamesStr := strings.Join(link.Attrs().AltNames, ",")
+		altNamesStr := strings.Join(altNames[idx+1:], ",")
+
+		if linkAltNamesStr != altNamesStr {
+			t.Fatalf("Expected %s AltNames, got %s", altNamesStr, linkAltNamesStr)
+		}
 	}
+
 }
 
 func TestLinkSetARP(t *testing.T) {

